.github/workflows/pytest-remote-data.yml

```diff
@@ -0,0 +1,110 @@
+# A secondary test job that only runs the iotools tests if explicitly requested
+# (for pull requests) or on a push to the master branch.
+# Because the iotools tests require GitHub secrets, we need to be careful about 
+# malicious PRs accessing the secrets and exposing them externally.
+#
+# We prevent this by only running this workflow when a maintainer has looked
+# over the PR's diff and verified that nothing malicious seems to be going on.
+# The maintainer then adds the "remote-data" label to the PR, which will then
+# trigger this workflow via the combination of the "on: ... types:"
+# and "if:" sections below.  The first restricts the workflow to only run when
+# a label is added to the PR and the second requires one of the PR's labels
+# is the "remote-data" label.  Technically this is slightly different from
+# triggering when the "remote-data" label is added, since it will also trigger
+# when "remote-data" is added, then later some other label is added.  Maybe
+# there's a better way to do this.
+#
+# But wait, you say!  Can't a malicious PR get around this by modifying
+# this workflow file and removing the label requirement?  I think the answer
+# is "no" as long as we trigger the workflow on "pull_request_target" instead
+# of the usual "pull_request".  The difference is what context the workflow
+# runs inside: "pull_request" runs in the context of the fork, where changes
+# to the workflow definition will take immediate effect, while "pull_request_target"
+# runs in the context of the main pvlib repository, where the original (non-fork)
+# workflow definition is used instead.  Of course by switching away from the fork's
+# context to keep our original workflow definitions, we're also keeping all the
+# original code, so the tests won't be run against the PR's new code.  To fix this
+# we explicitly check out the PR's code as the first step of the workflow.
+# This allows the job to run modified pvlib & pytest code, but only ever via
+# the original workflow file.
+# So long as a maintainer always verifies that the PR's code is not malicious prior to
+# adding the label and triggering this workflow, I think this should not present
+# a security risk.
+#
+# Note that this workflow can be triggered again by removing and re-adding the
+# "remote-data" label to the PR.
+#
+# Note also that "pull_request_target" is also the only way for the secrets
+# to be accessible in the first place.
+# 
+# Further reading:
+# - https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+# - https://github.community/t/can-workflow-changes-be-used-with-pull-request-target/178626/7
+
+name: pytest-remote-data
+
+on:
+  pull_request_target:
+    types: [labeled]
```

Comment:
@kanderso-nrel if we were to delete this line, then we wouldn't have to add and remove the remote-data label every time to trigger the tests.

For PRs that do NOT have the remote-data label, the remote tests would simply be skipped due to the if statement in line 66:
![image](https://user-images.githubusercontent.com/39184289/185410505-450f1c67-adf4-45b4-9ee7-de0d0ed23c36.png)

Maybe there are some security aspects that I'm not aware of.

Reply:
Having thought about this more, I think this was intentional on my part (see the comments at the top of the workflow file).  Removing this line would allow a scenario where someone can submit a benign PR to lure us into adding the `remote-data` label, then push another commit with malicious content and have the workflow run without our sign off.  The way it is now, the workflow cannot run without a maintainer explicitly triggering it.

Whether keeping that security hole plugged is worth the annoyance of repeated labeling is another question...

Reply:
PyPI's recent labeling of pvlib as a critical project has me feeling like we should err on the side of caution here. Tests should have a minimum of real API calls anyways - the vast majority should be mocked.